File: app/src/main/java/ua/te/hackathon/smartcity2015/ui/main/events/browse/BrowseEventsView.java
package ua.te.hackathon.smartcity2015.ui.main.events.browse;

import android.support.annotation.NonNull;

import java.util.List;

import ua.te.hackathon.smartcity2015.db.model.Event;
import ua.te.hackathon.smartcity2015.ui.base.mvp.PresenterView;

/**
 * @author victor
 * @since 2016-02-13
 */
public interface BrowseEventsView extends PresenterView {

  void showLoadingView();

  void hideLoadingView();

  void deliverEventList(@NonNull List<Event> list);

  void deliverLoadingError(String error);

}

File: app/src/main/kotlin/ua/te/hackathon/smartcity2015/ui/main/events/browse/BrowseEventsView.kt
package ua.te.hackathon.smartcity2015.ui.main.events.browse

import ua.te.hackathon.smartcity2015.db.model.Event
import ua.te.hackathon.smartcity2015.ui.base.mvp.PresenterView

/**
 * @author victor
 * *
 * @since 2016-02-13
 */
interface BrowseEventsView : PresenterView {

  fun showLoadingView()

  fun hideLoadingView()

  fun deliverEventList(list: List<Event>)

  fun deliverLoadingError(error: String)

}
